crates/core_simd/src/ops/assign.rs

```diff
@@ -0,0 +1,124 @@
+//! Assignment operators
+use super::*;
+use core::ops::{AddAssign, MulAssign}; // commutative binary op-assignment
+use core::ops::{BitAndAssign, BitOrAssign, BitXorAssign}; // commutative bit binary op-assignment
+use core::ops::{DivAssign, RemAssign, SubAssign}; // non-commutative binary op-assignment
+use core::ops::{ShlAssign, ShrAssign}; // non-commutative bit binary op-assignment
+
+// Arithmetic
+
+macro_rules! assign_ops {
+    ($(impl<T, U, const LANES: usize> $assignTrait:ident<U> for Simd<T, LANES>
+        where
+            Self: $trait:ident,
+        {
+            fn $assign_call:ident(rhs: U) {
+                $call:ident
+            }
+        })*) => {
+        $(impl<T, U, const LANES: usize> $assignTrait<U> for Simd<T, LANES>
+        where
+            Self: $trait<U, Output = Self>,
+            T: SimdElement,
+            LaneCount<LANES>: SupportedLaneCount,
+        {
+            #[inline]
+            fn $assign_call(&mut self, rhs: U) {
+                *self = self.$call(rhs);
+            }
+        })*
+    }
+}
+
+assign_ops! {
```

Comment:
I understand why you did it this way, but it would save something like 100 lines implementing it along the lines of:

```rust
assign_ops! {
   AddAssign::add_assign -> Add:add,
   MulAssign::mul_assign -> Mul::mul,
   ...
}
```

Using the rust-like syntax usually helps IMO but here the file would be only like 40 lines so it might actually be easier to read.

Reply:
(This probably applies to the other macros too, as long as they all stay in their own files to limit the scope of what you're looking at)

Reply:
I am not quiiite inclined to minimize it that far. I found the style of making the relationships in the macro more implicit slowed down finding what parts I needed to refactor out.
I _am_ inclined to minimize it in a way that still exposes all the relationships.
This needs a followup PR with more... housekeeping-y cleanup that isn't strictly related to this change anyways, so I will try to address this concern then.

Reply:
Rust will never compete with Julia or C/C++ in AI/ML if we don't push and demand the fastest iterations in the world.  We need everyone giving feedback and constant pursuit of excellence here.